src/smt/smt_engine.cpp

```diff
@@ -705,6 +709,35 @@ SmtEngine::SmtEngine(ExprManager* em)
 
 void SmtEngine::finishInit()
 {
+  // Notice that finishInit is called when options are finalized. If we are
+  // parsing smt2, this occurs at the moment we enter "Assert mode", page 52
+  // of SMT-LIB 2.6 standard.
+
+  // Inialize the resource manager based on the options.
+  d_resourceManager->setHardLimit(options::hardLimit());
+  if (options::perCallResourceLimit() != 0)
+  {
+    d_resourceManager->setResourceLimit(options::perCallResourceLimit(), false);
+  }
+  if (options::cumulativeResourceLimit() != 0)
+  {
+    d_resourceManager->setResourceLimit(options::cumulativeResourceLimit(),
+                                        true);
+  }
+  if (options::perCallMillisecondLimit() != 0)
+  {
+    d_resourceManager->setTimeLimit(options::perCallMillisecondLimit(), false);
+  }
+  if (options::cumulativeMillisecondLimit() != 0)
+  {
+    d_resourceManager->setTimeLimit(options::cumulativeMillisecondLimit(),
+                                    true);
+  }
+  if (options::cpuTime())
+  {
+    d_resourceManager->useCPUTime(true);
+  }
```

Comment:
One thing that we should consider here: Should we do those in `SmtEngine::setOption()` since they are options that could change after the engine has finished initializing? I think that's why we may have had these callbacks.

Reply:
Yes, I have (several) branches that attempt to refactor this which do what you suggest. Let's address on future PR.